changelog/fragments/k8s-1.24-bump.yaml

```diff
@@ -0,0 +1,110 @@
+# entries is a list of entries to include in
+# release notes and/or the migration guide
+
+# TODO(everettraven): update changelog
+entries:
+  - description: >
+      For Go (go/v3) and Helm Hybrid (hybrid) Operators: Upgrade Kubernetes 1.23 to 1.24
+
+    # kind is one of:
+    # - addition
+    # - change
+    # - deprecation
+    # - removal
+    # - bugfix
+    kind: "change"
+
+    # Is this a breaking change?
+    breaking: false
+
+    # Migration can be defined to automatically add a section to
+    # the migration guide. This is required for breaking changes.
+    migration:
+      header: Go (go/v3) and Hybrid Helm (hybrid) Operators - Upgrading to Kubernetes 1.24
+      body: |
+        To update your project to support Kubernetes 1.24:
+        
+        1. In `go.mod`:
+
+        - Replace `k8s.io/api v0.23.5` with `k8s.io/api v0.24.0`
+        - Replace `k8s.io/apimachinery v0.23.5` with  `k8s.io/apimachinery v0.24.0`
+        - Replace `k8s.io/client-go v0.23.5` with `k8s.io/client-go v0.24.0`
```

Comment:
This is not required because will be synced automatically when we bump controller-runtime but that is fine. Note that we have note them on the bolleplate by default